docs/CIP/CIP-XXXX/messages/client/subscribe.md

```diff
@@ -0,0 +1,201 @@
+                "type": "string"
+              }
+            },
+            "required": ["name", "network"]
+          },
+          "signature": {
+            "type": "string",
+            "pattern": "^[A-Za-z0-9+/=]*$"
+          },
+          /* <blockchain specific grouped fields> */
+          "cardano": {
+            "type": "object",
+            "properties": {
+              "credentials": {
+                "type": "object",
+                "properties": {
+                  "payment": {
+                    "type": "array",
+                    "items": {
+                      "type": "string",
+                      "pattern": "^[A-Za-z0-9+/=]*$"
+                    }
+                  },
+                  "stake": {
+                    "type": "array",
+                    "items": {
+                      "type": "string",
+                      "pattern": "^[A-Za-z0-9+/=]*$"
+                    }
+                  }
+                },
+                "required": ["payment", "stake"]
+              },
+              "extensions": {
+                "type": "object",
+                "properties": {
+                  "resolveTxInput": {
+                    "type": "boolean",
+                    "default": false
+                  },
+                  "assetMetadata": {
+                    "type": "boolean",
+                    "default": false
+                  }
+                }
+              }
+            },
+            "required": ["credentials", "extensions"]
+          }
+        },
+        "required": ["blockchain", "signature"]
+      }
+    },
+    "timestamp": {
+      "type": "string",
+      "format": "date-time"
+    }
+  },
+  "required": ["type", "topics", "timestamp"],
+  "additionalProperties": false
+}
+```
+
+The `signature` field is generated by signing a SHA256 HMAC hashed string. The prehash string is constructed by concatenating the subscribed `blockchains` + `timestamp`.
+
+> [!NOTE]
+> For subscribing to multiple blockchains, the prehash string must be generated for each as `<blockchain.name>.<blockchain.network>`, for example:
+> `cardano.mainnet+2024-06-27T12:34:56Z`.
+
+Apply a SHA256 HMAC using either the payment signing key or staking signing key, and then base64-encode it as final payload within the initial message. Pass the output of this SHA256 HMAC to the signature field.
+
+The signature verifies ownership of the private key, whose corresponding public key is used to filter relevant block transactions.
+
+## Subscribe Example
+
+```json
+{
+  "type": "subscribe",
+  "topics": [
+    {
+      "blockchain": { "name": "cardano", "network": "mainnet" },
+      "version": ["0.1"],
+      "signature": "OGNiOWIyNGVjOTMxZmY3N2MzYjQxOTY3OWE0YTcwMzczZmVkZmIxNDZmMDE0ODk0Nzg4YjUxMmIzMjE4MDdiYw==", // base64, SHA256 HMAC with your signing key
+      "cardano": {
+        "credentials": {
+          "extendedPublicKey": ["acct_xvk..."] // this field follows  CIP-0005
```

Comment:
'credentials' in Cardano refer to payment/stake key hash or script hash used for addresses. We probably shouldn't classify xpub as a credential
```suggestion
      "cardano": {
        "extendedPublicKey": ["acct_xvk..."] // this field follows  CIP-0005
        "credentials": {
```

Reply:
Probably here I'm still missing something.

Which is the relationship between credentials and public keys?

- Isn't there a 1 to 1 relationship?
- Can we compute the one from the other with a public process?

Reply:
Credential can be either:
- key hash
- script hash

So,
> Isn't there a 1 to 1 relationship?
Nope

> Can we compute the one from the other with a public process?
Given the key, you can hash it to get the credential. Not the other way around.